File: Arrays/Reverse_array.cpp
#include<bits/stdc++.h>
using namespace std;

void ReverseArray(int arr[],int n){
    int start = 0;
    int end = n-1;
    int temp;
    while(start<end){
        temp = arr[start];
        arr[start] = arr[end];
        arr[end] = temp;

        start++;
        end--;
    }
}


void PrintArray(int arr[],int n){

    for(int i=0;i<n;i++){
        cout<<arr[i]<<" ";
    }
    cout<<endl;
}

int main() {
 
int array[] = {1,3,55,67,890,434};
int n = sizeof(array)/sizeof(array[0]);

ReverseArray(array,n);
PrintArray(array,n);

return 0;
}

File: Arrays/Reverse_array.java

class Reverse_array {

    void reverse(int[] arr, int size) {
        int temp;
        int left = 0;
        int right = size - 1;
        while (left < right) {
            temp = arr[left];
            arr[left] = arr[right];
            arr[right] = temp;

            left++;
            right--;
        }

        for (int i = 0; i < size; i++) {
            System.out.print(arr[i] + " ");
        }
    }

    public static void main(String[] args) {
        int[] arr = new int[]{1, 2, 3, 4, 5, 6, 7};
        Reverse_array obj = new Reverse_array();

        int size = 7;

        obj.reverse(arr, size);

    }
}
